docs_src/microservices/application/ContextAPI.md

```diff
@@ -1,67 +1,94 @@
+    // unneeded HTTP calls to Core Metadata
+    GetDeviceResource(profileName string, resourceName string) (dtos.DeviceResource, error)
+    // AddValue stores a value for access within other functions in pipeline
+    AddValue(key string, value string)
+    // RemoveValue deletes a value stored in the context at the given key
+    RemoveValue(key string)
+    // GetValue attempts to retrieve a value stored in the context at the given key
+    GetValue(key string) (string, bool)
+    // GetAllValues returns a read-only copy of all data stored in the context
+    GetAllValues() map[string]string
+    // ApplyValues looks in the provided string for placeholders of the form
+    // '{any-value-key}' and attempts to replace with the value stored under
+    // the key in context storage.  An error will be returned if any placeholders
+    // are not matched to a value in the context.
+    ApplyValues(format string) (string, error)
 }
 ```
 
 ## Clients
 
-### LoggingClient
+### LoggingClient()
 
 The `LoggingClient` exposed on the context is available to leverage logging libraries/service utilized throughout the EdgeX framework. The SDK has initialized everything so it can be used to log `Trace`, `Debug`, `Warn`, `Info`, and `Error` messages as appropriate. See [simple-filter-xml/main.go](https://github.com/edgexfoundry-holding/app-service-examples/blob/master/app-services/simple-filter-xml/main.go) for an example of how to use the `LoggingClient`.
 
-### EventClient 
+### EventClient()
 
 The `EventClient ` exposed on the context is available to leverage Core Data's `Event` API. See [interface definition](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/coredata/event.go#L35) for more details. This client is useful for querying events and is used by the [PushToCore](#pushtocore) convenience API described below.
 
-### ValueDescriptorClient
-
-The `ValueDescriptorClient ` exposed on the context is available to leverage Core Data's `ValueDescriptor` API. See [interface definition](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/coredata/value_descriptor.go#L29) for more details. Useful for looking up the value descriptor for a reading received.
-
-### CommandClient 
+### CommandClient() 
 
 The `CommandClient ` exposed on the context is available to leverage Core Command's `Command` API. See [interface definition](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/command/client.go#L28) for more details. Useful for sending commands to devices.
 
-### NotificationsClient
+### NotificationsClient()
 
 The `NotificationsClient` exposed on the context is available to leverage Support Notifications' `Notifications` API. See [README](https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/notifications/README.md) for more details. Useful for sending notifications. 
```

Comment:
Update link to https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/clients/interfaces/notification.go